ql/src/java/org/apache/hadoop/hive/ql/plan/ExprNodeDescUtils.java

```diff
@@ -233,6 +235,23 @@ public static ExprNodeGenericFuncDesc and(List<ExprNodeDesc> exps) {
     return new ExprNodeGenericFuncDesc(TypeInfoFactory.booleanTypeInfo, new GenericUDFOPAnd(), "and", flatExps);
   }
 
+  /**
+   * Create an expression for computing a hash by recursively hashing given expressions by two:
+   * <pre>
+   * Input: HASH(A, B, C, D)
+   * Output: HASH(HASH(HASH(A,B),C),D)
+   * </pre>
+   */
+  public static ExprNodeGenericFuncDesc hash(List<ExprNodeDesc> exps) {
+    assert exps.size() >= 2;
+    ExprNodeDesc hashExp = exps.get(0);
+    for (int i = 1; i < exps.size(); i++) {
+      List<ExprNodeDesc> hArgs = Arrays.asList(hashExp, exps.get(i));
+      hashExp = new ExprNodeGenericFuncDesc(TypeInfoFactory.intTypeInfo, new GenericUDFMurmurHash(), "hash", hArgs);
```

Comment:
it seems like we have some inconsistency in `GenericUDFMurmurHash` which is registered as `murmur_hash` in the `FunctionRegistry` ; however in the UDF's annotation it only has `hash` - and here as well we use simply "hash".
a change like this will most likely cause a lot of q.out changes - could you file a follow-up ticket?

Reply:
Good catch @kgyrtkirk ! I've never noticed that we have two different UDFs for hashing. Indeed having the same annotation can create quite some confusion and difficult to debug problems. I guess your suggestion is to change the annotation of GenericUDFMurmurHash to murmur_hash right?

Reply:
yes; excatly

Reply:
Logged https://issues.apache.org/jira/browse/HIVE-24572 for this purpose.